service/src/main/java/greencity/service/HabitServiceImpl.java

```diff
@@ -28,56 +32,154 @@
         Page<HabitTranslation> pages =
             habitTranslationRepo.findAllByLanguageCode(pageable, language);
-        List<HabitTranslationDto> habitTranslationDtos =
+        List<HabitDto> habitTranslationDtos =
             pages.stream()
-                .map(habit -> modelMapper.map(habit, HabitTranslationDto.class))
+                .map(habitTranslation -> modelMapper.map(habitTranslation, HabitDto.class))
                 .collect(Collectors.toList());
         return new PageableDto<>(habitTranslationDtos, pages.getTotalElements(),
             pages.getPageable().getPageNumber(),
             pages.getTotalPages());
     }
-}
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional
+    public HabitManagementDto saveHabitAndTranslations(HabitManagementDto habitManagementDto, MultipartFile image) {
+        Habit habit = habitRepo.save(Habit.builder()
+            .image(habitManagementDto.getImage())
+            .habitTranslations(
+                habitManagementDto.getHabitTranslations().stream()
+                    .map(translationDto -> HabitTranslation.builder()
+                        .description(translationDto.getDescription())
+                        .habitItem(translationDto.getHabitItem())
+                        .name(translationDto.getName())
+                        .language(modelMapper.map(
+                            languageService.findByCode(translationDto.getLanguageCode()),
+                            Language.class)).build())
+                    .collect(Collectors.toList())
+            ).build());
+        habit.getHabitTranslations().forEach(ht -> ht.setHabit(habit));
+
+        uploadImageForHabit(habitManagementDto, image, habit);
+
+        habitTranslationRepo.saveAll(habit.getHabitTranslations());
+        return modelMapper.map(habit, HabitManagementDto.class);
+    }
+
+    private void uploadImageForHabit(HabitManagementDto habitManagementDto, MultipartFile image, Habit habit) {
+        if (habitManagementDto.getImage() != null) {
+            image = fileService.convertToMultipartImage(habitManagementDto.getImage());
+        }
+        if (image != null) {
+            habit.setImage(fileService.upload(image).toString());
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional
+    public void update(HabitManagementDto habitManagementDto, MultipartFile image) {
+        Habit habit = habitRepo.findById(habitManagementDto.getId())
+            .orElseThrow(() -> new WrongIdException(ErrorMessage.HABIT_NOT_FOUND_BY_ID));
+
+        habit.getHabitTranslations()
+            .forEach(ht -> {
+                HabitTranslationManagementDto htmd = habitManagementDto.getHabitTranslations().stream()
+                    .filter(e -> e.getLanguageCode().equals(ht.getLanguage().getCode())).findFirst()
+                    .orElseThrow(RuntimeException::new);
```

Comment:
Better move this from lambda. Make getHabitTranslations rerutn Map<String, HabitTranslationManagementDto>, where key will be lamguage. Than find needed HabitTranslationManagementDto using get in Map.

Reply:
Also move all setting values to private method

Reply:
Created new card with issue.